app/models/katello/content_view_environment.rb

```diff
@@ -28,11 +28,20 @@ class ContentViewEnvironment < Katello::Model
 
     scope :non_default, -> { joins(:content_view).where("katello_content_views.default" => false) }
     scope :default, -> { joins(:content_view).where("katello_content_views.default" => true) }
+    scope :non_generated, -> { where(content_view: ::Katello::ContentView.ignore_generated) }
```

Comment:
When you do `ContentViewEnvironment.all` it will run it through the default scope. So `ContentViewEnvironment.all` would be equivalent to `ContentViewEnvironment.where(content_view: ::Katello::ContentView.ignore_generated)`

and you'd have to do `ContentViewEnvironment.unscoped` if you don't want it limited.

Reply:
Generated CVEs get promoted to library. Its just a secret publish. We just dont show em in the UI

Reply:
Default Scope has its own set of problems. We'll have to remember to do `unscoped` where ever we think about generated. I'd rather have it show up in the result set.